podpac/core/interpolation/nearest_neighbor_interpolator.py

```diff
@@ -80,8 +80,8 @@ def interpolate(self, udims, source_coordinates, source_data, eval_coordinates,
         def is_stacked(d):
             return "_" in d
 
-        if hasattr(source_data, "attrs"):
-            bounds = source_data.attrs.get("bounds", {d: None for d in source_coordinates.udims})
+        if hasattr(source_data, "attrs") and "bounds" in source_data.attrs:
+            bounds = source_data.attrs["bounds"]
```

Comment:
This will/should break something. You need to initialize bounds in an else branch if you want to eliminate the 'dense' statement. 
```suggestion
            bounds = source_data.attrs["bounds"]
        else:
            bounds = {d: None for d in source_coordinates.udims}          
```

Reply:
The else statement is already is there!

Reply:
Oh well.. I suppose that's what happens when you only look at the diffs. 😝